galsim/correlatednoise.py

```diff
@@ -1465,3 +1504,86 @@ def __repr__(self):
+    @param SEDs    SEDs of associated ChromaticSum components.
+    """
+    def __init__(self, Sigma, SEDs):
+        self.Sigma = Sigma
+        self.SEDs = SEDs
+
+    def __mul__(self, variance_ratio):
+        return self.withScaledVariance(variance_ratio)
+
+    def transform(self, dudx, dudy, dvdx, dvdy):
+        Sigma = {}
+        for k, v in iteritems(self.Sigma):
+            Sigma[k] = v.transform(dudx, dudy, dvdx, dvdy)
+        return CovarianceSpectrum(Sigma, self.SEDs)
+
+    def withScaledVariance(self, variance_ratio):
+        Sigma = {}
+        for k, v in iteritems(self.Sigma):
+            Sigma[k] = v * variance_ratio
+        return CovarianceSpectrum(Sigma, self.SEDs)
+
+    def toNoise(self, bandpass, PSF, wcs, rng=None):
+        """ Derive the galsim.CorrelatedNoise object for the associated ChromaticSum when convolved
+        with `PSF` and drawn through `bandpass` onto pixels with specified `wcs`.
+
+        @param bandpass   galsim.Bandpass object representing filter image is drawn through.
+        @param PSF        output chromatic PSF to convolve by.
+        @param wcs        WCS of output pixel scale.
+        @param rng        Random number generator to forward to resulting CorrelatedNoise object.
+        @returns  CorrelatedNoise object.
+        """
+        import numpy as np
+        NSED = len(self.SEDs)
+        maxk = np.min([PSF.evaluateAtWavelength(bandpass.blue_limit).maxK(),
+                       PSF.evaluateAtWavelength(bandpass.red_limit).maxK()])
+        stepk = np.max([PSF.evaluateAtWavelength(bandpass.blue_limit).stepK(),
+                        PSF.evaluateAtWavelength(bandpass.red_limit).stepK()])
+        nk = 2*int(np.ceil(maxk/stepk))
+
+        PSF_eff_kimgs = np.empty((NSED, nk, nk), dtype=np.complex128)
+        for i, sed in enumerate(self.SEDs):
+            # Assume that PSF does not yet include pixel contribution, so add it in.
+            conv = galsim.Convolve(PSF, galsim.Pixel(wcs.scale)) * sed
+            PSF_eff_kimgs[i] = conv.drawKImage(bandpass, nx=nk, ny=nk, scale=stepk).array
+        pkout = np.zeros((nk, nk), dtype=np.float64)
+        for i in range(NSED):
+            for j in range(i, NSED):
+                s = self.Sigma[(i, j)].drawKImage(nx=nk, ny=nk, scale=stepk).array
+                pkout += (np.conj(PSF_eff_kimgs[i]) * s * PSF_eff_kimgs[j] *
+                          (2 if i != j else 1)).real
+        pk = galsim.ImageCD(pkout + 0j, scale=stepk)  # imag part should be zero
+        iki = galsim.InterpolatedKImage(pk)
+        iki *= wcs.pixelArea()**2  # determined this empirically
+        return galsim.correlatednoise._BaseCorrelatedNoise(rng, iki, wcs)
+
+    def __eq__(self, other):
+        return (isinstance(other, galsim.CovarianceSpectrum) and
+                self.SEDs == other.SEDs and
+                self.Sigma == other.Sigma)
+    def __ne__(self, other): return not self.__eq__(other)
```

Comment:
This isn't tested.  Should probably add a quick test that `covspec1 != covspec2` for two different CovarianceSpectra.